### PR TITLE
Vet clinic database: query and update animals table

### PR DESCRIPTION
**Pull request by JEFFKAY16**
## Vet clinic database: query and update animals table
In this update I did the following: 

- Add a column `species` of type `string` to your `animals` table. Modify your `schema.sql` file.
- Insert the following data:
  - Animal: His name is Charmander. He was born on Feb 8th, 2020, and currently weighs -11kg. He is not neutered and he has never tried to escape.
  - Animal: Her name is Plantmon. She was born on Nov 15th, 2021, and currently weighs -5.7kg. She is neutered and she has tried to escape 2 times.
  - Animal: His name is Squirtle. He was born on Apr 2nd, 1993, and currently weighs -12.13kg. He was not neutered and he has tried to escape 3 times.
  - Animal: His name is Angemon. He was born on Jun 12th, 2005, and currently weighs -45kg. He is neutered and he has tried to escape once.
  - Animal: His name is Boarmon. He was born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried to escape 7 times.
  - Animal: Her name is Blossom. She was born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried to escape 3 times.
  - Animal: His name is Ditto. He was born on May 14th, 2022, and currently weighs 22kg. He is neutered and he has tried to escape 4 times.
- Remember to add these insertions to your `data.sql` file.
- Inside a transaction update the `animals` table by setting the `species` column to `unspecified`. Verify that change was made. Then roll back the change and verify that `species` columns went back to the state before transaction.
- Inside a transaction:
  - Update the `animals` table by setting the `species` column to `digimon` for all animals that have a name ending in `mon`.
  - Update the `animals` table by setting the `species` column to `pokemon` for all animals that don't have `species` already set.
  - Commit the transaction.
  - Verify that change was made and persists after commit.
- Now, take a deep breath and... **Inside a transaction** delete all records in the `animals` table, then roll back the transaction.
- After the roll back verify if all records in the `animals` table still exist. After that you can start breath as usual ;) 
- Inside a transaction:
  - Delete all animals born after Jan 1st, 2022.
  - Create a savepoint for the transaction.
  - Update all animals' weight to be their weight multiplied by -1.
  - Rollback to the savepoint
  - Update all animals' weights that are negative to be their weight multiplied by -1.
  - Commit transaction
- Write queries to answer the following questions:
  - How many animals are there?
  - How many animals have never tried to escape?
  - What is the average weight of animals?
  - Who escapes the most, neutered or not neutered animals?
  - What is the minimum and maximum weight of each type of animal?
  - What is the average number of escape attempts per animal type of those born between 1990 and 2000?
- Remember to add these queries to your `queries.sql` file.

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -1,17 +1,87 @@
...
+
+-- Who escapes the most, neutered or not neutered animals?
+SELECT name, MAX(escape_attempts) as max FROM animals GROUP BY neutered, name ORDER BY max DESC LIMIT 1;
+
+-- What is the minimum and maximum weight of each type of animal?
+SELECT species, MAX(weight_kg) as max_weigth, MIN(weight_kg) AS min_weight FROM animals GROUP by species;
 
-SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
\ No newline at end of file
+SELECT species, AVG(escape_attempts) AS avg_escapes  FROM animals WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-01-01' GROUP BY species;
```
Comment (mmsesay) [inline queries.sql]: - [x] If we think about animals that are born after the first day and first month of the year 2000. This query will not fetch them don't you think so?
I will suggest you end your check at the last day and last month of the year 2000.

See example:
```sql
SELECT species, AVG(escape_attempts) FROM animals
GROUP BY species
WHERE  date_of_birth >= '1990/01/01' AND date_of_birth <= '2000/31/12';
```

File: schema.sql [hunk 2]
```diff
@@ -1,20 +1,19 @@
...
+    neutered BOOLEAN,   
+    weight_kg DECIMAL
+);
 
-/* Created table */
+INSERT INTO animals (name,date_of_birth,escape_attempts,neutered,weight_kg) VALUES
+('Agumon','2020-02-03',0,true,'10.23'),
+('Gabumon','2018-11-15',2,true,'8'),
+('Pikachu', '2021-01-07',1,false, '15.04'),
+('Devimon', '2017-05-12', 5, true, '11');
```
Comment (mmsesay) [inline schema.sql]: - [x] Please remove these lines of code from this file.  Kindly note that Everything that populates tables with data should be saved in the data.sql file.

**Review by mmsesay — CHANGES_REQUESTED** [review]
Hi  @JEFFKAY16,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 1 above)
Comment (JEFFKAY16) [reply queries.sql]: Done, thank you

File: schema.sql (hunk 2 above)
Comment (JEFFKAY16) [reply schema.sql]: Done, thanks.

**Review by mmsesay — APPROVED** [review]
Hi @JEFFKAY16,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**